inlong-sort/sort-connectors/iceberg/src/main/java/org/apache/inlong/sort/iceberg/sink/multiple/WholeDatabaseMigrationOperator.java

```diff
@@ -0,0 +1,228 @@
+            properties.put("format-version", "2"); // todo:后续考虑默认参数给哪些，并且将这个默认参数暴露在表参数上
+            properties.put("write.upsert.enabled", "true");
+            // 设置了这个属性自动建表后hive才能查询到
+            properties.put("engine.hive.enabled", "true");
+
+            try {
+                catalog.createTable(tableId, schema, PartitionSpec.unpartitioned(), properties.build());
+                LOG.info("Auto create Table({}) in Database({}) in Catalog({})!",
+                        tableId.name(), tableId.namespace(), catalog.name());
+            } catch (AlreadyExistsException e) {
+                LOG.warn("Table({}) already exist in Database({}) in Catalog({})!",
+                        tableId.name(), tableId.namespace(), catalog.name());
+            }
+        }
+
+        handleSchemaInfoEvent(tableId, catalog.loadTable(tableId).schema());
+    }
+
+    private void handldAlterSchemaEventFromOperator(TableIdentifier tableId, Schema oldSchema, Schema newSchema) {
+        Table table = catalog.loadTable(tableId);
+
+        // The transactionality of changes is guaranteed by comparing the old schema with the current schema of the
+        // table.
+        // Judging whether changes can be made by schema comparison (currently only column additions are supported),
+        // for scenarios that cannot be changed, it is always considered that there is a problem with the data.
+        Transaction transaction = table.newTransaction();
+        if (table.schema().sameSchema(oldSchema)) {
+            List<TableChange> tableChanges = TableChange.diffSchema(oldSchema, newSchema);
+            TableChange.applySchemaChanges(transaction.updateSchema(), tableChanges);
+            LOG.info("Schema evolution in table({}) for table change: {}", tableId, tableChanges);
+        }
+        transaction.commitTransaction();
+        handleSchemaInfoEvent(tableId, table.schema());
+    }
+
+    // =============================== 工具方法 =================================================================
+    // The way to judge compatibility is whether all the field names in the old schema exist in the new schema
+    private boolean isCompatible(Schema newSchema, Schema oldSchema) {
+        for (NestedField field : oldSchema.columns()) {
+            if (newSchema.findField(field.name()) == null) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    // 从数据中解析schema信息并转换成为flink内置的schema,对不同的格式（canal-json、ogg）以插件接口的方式提供这个转换方式
+    private RecordWithSchema parseRecord(JsonNode data) throws IOException {
+        String databaseStr = dynamicSchemaFormat.parse(data, multipleSinkOption.getDatabasePattern());
+        String tableStr = dynamicSchemaFormat.parse(data, multipleSinkOption.getTablePattern());
+        List<String> pkListStr = dynamicSchemaFormat.extractPrimaryKeyNames(data);
+        RowType schema = dynamicSchemaFormat.extractSchema(data, pkListStr);

```

Comment:
Compatibility processing for iceberg-specific data types based schema or add a callback function in the 'dynamicSchemaFormat.extractSchema'?

Reply:
already process it after scheam change